### PR TITLE
Maven: make jdk11 latest

### PR DESCRIPTION
**Pull request by carlossg**
Fixes https://github.com/carlossg/docker-maven/issues/93

### CODE REVIEW

**Comment by yosifkit** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index 07f6607..cc0d973 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -1,3 +1,8 @@
+maven:3-ibmjava-8 @ amd64
+maven:3-ibmjava-8 @ i386
+maven:3-ibmjava-8 @ ppc64le
+maven:3-ibmjava-8 @ s390x
+maven:3-ibmjava-8-alpine @ amd64
 maven:3-jdk-7 @ amd64
 maven:3-jdk-7 @ arm32v5
 maven:3-jdk-7 @ arm32v7
@@ -12,6 +17,26 @@ maven:3-jdk-7-slim @ amd64
 maven:3-jdk-7-slim @ arm32v5
 maven:3-jdk-7-slim @ arm32v7
 maven:3-jdk-7-slim @ i386
+maven:3-jdk-8 @ amd64
+maven:3-jdk-8 @ arm32v5
+maven:3-jdk-8 @ arm32v7
+maven:3-jdk-8 @ arm64v8
+maven:3-jdk-8 @ i386
+maven:3-jdk-8 @ ppc64le
+maven:3-jdk-8 @ s390x
+maven:3-jdk-8-alpine @ amd64
+maven:3-jdk-8-alpine @ arm32v6
+maven:3-jdk-8-alpine @ arm64v8
+maven:3-jdk-8-alpine @ i386
+maven:3-jdk-8-alpine @ ppc64le
+maven:3-jdk-8-alpine @ s390x
+maven:3-jdk-8-slim @ amd64
+maven:3-jdk-8-slim @ arm32v5
+maven:3-jdk-8-slim @ arm32v7
+maven:3-jdk-8-slim @ arm64v8
+maven:3-jdk-8-slim @ i386
+maven:3-jdk-8-slim @ ppc64le
+maven:3-jdk-8-slim @ s390x
 maven:3-jdk-10 @ amd64
 maven:3-jdk-10 @ arm32v5
 maven:3-jdk-10 @ arm32v7
@@ -26,33 +51,8 @@ maven:3-jdk-10-slim @ arm64v8
 maven:3-jdk-10-slim @ i386
 maven:3-jdk-10-slim @ ppc64le
 maven:3-jdk-10-slim @ s390x
-maven:3-jdk-11 @ amd64
-maven:3-jdk-11 @ arm32v5
-maven:3-jdk-11 @ arm32v7
-maven:3-jdk-11 @ arm64v8
-maven:3-jdk-11 @ i386
-maven:3-jdk-11 @ ppc64le
-maven:3-jdk-11 @ s390x
-maven:3-jdk-11-slim @ amd64
-maven:3-jdk-11-slim @ arm32v5
-maven:3-jdk-11-slim @ arm32v7
-maven:3-jdk-11-slim @ arm64v8
-maven:3-jdk-11-slim @ i386
-maven:3-jdk-11-slim @ ppc64le
-maven:3-jdk-11-slim @ s390x
 maven:3-jdk-12 @ amd64
 maven:3-jdk-12-alpine @ amd64
-maven:alpine @ amd64
-maven:alpine @ arm32v6
-maven:alpine @ arm64v8
-maven:alpine @ i386
-maven:alpine @ ppc64le
-maven:alpine @ s390x
-maven:ibmjava @ amd64
-maven:ibmjava @ i386
-maven:ibmjava @ ppc64le
-maven:ibmjava @ s390x
-maven:ibmjava-alpine @ amd64
 maven:latest @ amd64
 maven:latest @ arm32v5
 maven:latest @ arm32v7
diff --git a/_bashbrew-list b/_bashbrew-list
index 7dc10ce..89cc1b3 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -1,5 +1,4 @@
 maven:3
-maven:3-ibmjava
 maven:3-ibmjava-8
 maven:3-ibmjava-8-alpine
 maven:3-jdk-7
@@ -15,11 +14,8 @@ maven:3-jdk-11-slim
 maven:3-jdk-12
 maven:3-jdk-12-alpine
 maven:3.6
-maven:3.6-alpine
-maven:3.6-ibmjava
 maven:3.6-ibmjava-8
 maven:3.6-ibmjava-8-alpine
-maven:3.6-ibmjava-alpine
 maven:3.6-jdk-7
 maven:3.6-jdk-7-alpine
 maven:3.6-jdk-7-slim
@@ -34,11 +30,8 @@ maven:3.6-jdk-12
 maven:3.6-jdk-12-alpine
 maven:3.6-slim
 maven:3.6.0
-maven:3.6.0-alpine
-maven:3.6.0-ibmjava
 maven:3.6.0-ibmjava-8
 maven:3.6.0-ibmjava-8-alpine
-maven:3.6.0-ibmjava-alpine
 maven:3.6.0-jdk-7
 maven:3.6.0-jdk-7-alpine
 maven:3.6.0-jdk-7-slim
@@ -52,8 +45,5 @@ maven:3.6.0-jdk-11-slim
 maven:3.6.0-jdk-12
 maven:3.6.0-jdk-12-alpine
 maven:3.6.0-slim
-maven:alpine
-maven:ibmjava
-maven:ibmjava-alpine
 maven:latest
 maven:slim
diff --git a/maven_ibmjava-alpine/Dockerfile b/maven_3-ibmjava-8-alpine/Dockerfile
similarity index 100%
rename from maven_ibmjava-alpine/Dockerfile
rename to maven_3-ibmjava-8-alpine/Dockerfile
diff --git a/maven_3-jdk-11-slim/mvn-entrypoint.sh b/maven_3-ibmjava-8-alpine/mvn-entrypoint.sh
similarity index 100%
rename from maven_3-jdk-11-slim/mvn-entrypoint.sh
rename to maven_3-ibmjava-8-alpine/mvn-entrypoint.sh
diff --git a/maven_3-jdk-11-slim/settings-docker.xml b/maven_3-ibmjava-8-alpine/settings-docker.xml
similarity index 100%
rename from maven_3-jdk-11-slim/settings-docker.xml
rename to maven_3-ibmjava-8-alpine/settings-docker.xml
diff --git a/maven_ibmjava/Dockerfile b/maven_3-ibmjava-8/Dockerfile
similarity index 100%
rename from maven_ibmjava/Dockerfile
rename to maven_3-ibmjava-8/Dockerfile
diff --git a/maven_3-jdk-11/mvn-entrypoint.sh b/maven_3-ibmjava-8/mvn-entrypoint.sh
similarity index 100%
rename from maven_3-jdk-11/mvn-entrypoint.sh
rename to maven_3-ibmjava-8/mvn-entrypoint.sh
diff --git a/maven_3-jdk-11/settings-docker.xml b/maven_3-ibmjava-8/settings-docker.xml
similarity index 100%
rename from maven_3-jdk-11/settings-docker.xml
rename to maven_3-ibmjava-8/settings-docker.xml
diff --git a/maven_3-jdk-11-slim/Dockerfile b/maven_3-jdk-11-slim/Dockerfile
deleted file mode 100644
index ee03040..0000000
diff --git a/maven_3-jdk-11/Dockerfile b/maven_3-jdk-11/Dockerfile
deleted file mode 100644
index 5b8ade4..0000000
diff --git a/maven_alpine/Dockerfile b/maven_3-jdk-8-alpine/Dockerfile
similarity index 100%
rename from maven_alpine/Dockerfile
rename to maven_3-jdk-8-alpine/Dockerfile
diff --git a/maven_alpine/mvn-entrypoint.sh b/maven_3-jdk-8-alpine/mvn-entrypoint.sh
similarity index 100%
rename from maven_alpine/mvn-entrypoint.sh
rename to maven_3-jdk-8-alpine/mvn-entrypoint.sh
diff --git a/maven_alpine/settings-docker.xml b/maven_3-jdk-8-alpine/settings-docker.xml
similarity index 100%
rename from maven_alpine/settings-docker.xml
rename to maven_3-jdk-8-alpine/settings-docker.xml
diff --git a/maven_slim/Dockerfile b/maven_3-jdk-8-slim/Dockerfile
similarity index 100%
copy from maven_slim/Dockerfile
copy to maven_3-jdk-8-slim/Dockerfile
diff --git a/maven_ibmjava-alpine/mvn-entrypoint.sh b/maven_3-jdk-8-slim/mvn-entrypoint.sh
similarity index 100%
rename from maven_ibmjava-alpine/mvn-entrypoint.sh
rename to maven_3-jdk-8-slim/mvn-entrypoint.sh
diff --git a/maven_ibmjava-alpine/settings-docker.xml b/maven_3-jdk-8-slim/settings-docker.xml
similarity index 100%
rename from maven_ibmjava-alpine/settings-docker.xml
rename to maven_3-jdk-8-slim/settings-docker.xml
diff --git a/maven_latest/Dockerfile b/maven_3-jdk-8/Dockerfile
similarity index 100%
copy from maven_latest/Dockerfile
copy to maven_3-jdk-8/Dockerfile
diff --git a/maven_ibmjava/mvn-entrypoint.sh b/maven_3-jdk-8/mvn-entrypoint.sh
similarity index 100%
rename from maven_ibmjava/mvn-entrypoint.sh
rename to maven_3-jdk-8/mvn-entrypoint.sh
diff --git a/maven_ibmjava/settings-docker.xml b/maven_3-jdk-8/settings-docker.xml
similarity index 100%
rename from maven_ibmjava/settings-docker.xml
rename to maven_3-jdk-8/settings-docker.xml
diff --git a/maven_latest/Dockerfile b/maven_latest/Dockerfile
index 32c2f3c..5b8ade4 100644
--- a/maven_latest/Dockerfile
+++ b/maven_latest/Dockerfile
@@ -1,10 +1,15 @@
-FROM openjdk:8-jdk
+FROM openjdk:11-jdk
 
 ARG MAVEN_VERSION=3.6.0
 ARG USER_HOME_DIR="/root"
 ARG SHA=fae9c12b570c3ba18116a4e26ea524b29f7279c17cbaadc3326ca72927368924d9131d11b9e851b8dc9162228b6fdea955446be41207a5cfc61283dd8a561d2f
 ARG BASE_URL=https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries
 
+# Maven fails with 'Can't read cryptographic policy directory: unlimited'
+# because it looks for $JAVA_HOME/conf/security/policy/unlimited but it is in
+# /etc/java-9-openjdk/security/policy/unlimited
+RUN ln -s /etc/java-11-openjdk /usr/lib/jvm/java-11-openjdk-$(dpkg --print-architecture)/conf
+
 RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
   && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
   && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha512sum -c - \
diff --git a/maven_slim/Dockerfile b/maven_slim/Dockerfile
index 6904378..ee03040 100644
--- a/maven_slim/Dockerfile
+++ b/maven_slim/Dockerfile
@@ -1,4 +1,4 @@
-FROM openjdk:8-jdk-slim
+FROM openjdk:11-jdk-slim
 
 ARG MAVEN_VERSION=3.6.0
 ARG USER_HOME_DIR="/root"
@@ -10,6 +10,11 @@ RUN apt-get update && \
       curl procps \
   && rm -rf /var/lib/apt/lists/*
 
+# Maven fails with 'Can't read cryptographic policy directory: unlimited'
+# because it looks for $JAVA_HOME/conf/security/policy/unlimited but it is in
+# /etc/java-9-openjdk/security/policy/unlimited
+RUN ln -s /etc/java-11-openjdk /usr/lib/jvm/java-11-openjdk-$(dpkg --print-architecture)/conf
+
 RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
   && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
   && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha512sum -c - \
```

</details>

File: library/maven [hunk 1]
```diff
@@ -46,27 +46,27 @@ Architectures: amd64, arm32v5, arm32v7, i386
...
 Directory: jdk-8
 
-Tags: 3.6.0-ibmjava-8-alpine, 3.6.0-ibmjava-alpine, 3.6-ibmjava-8-alpine, 3.6-ibmjava-alpine, 3-ibmjava-8-alpine, ibmjava-alpine
+Tags: 3.6.0-ibmjava-8-alpine, 3.6-ibmjava-8-alpine, 3-ibmjava-8-alpine
 Architectures: amd64
 GitCommit: 05f4802aa5c253dcf75fe967c6f45b3fb1e2f26e
 Directory: ibmjava-8-alpine
 
-Tags: 3.6.0-ibmjava-8, 3.6.0-ibmjava, 3.6-ibmjava-8, 3.6-ibmjava, 3-ibmjava-8, 3-ibmjava, ibmjava
+Tags: 3.6.0-ibmjava-8, 3.6-ibmjava-8, 3-ibmjava-8
```
Comment (yosifkit) [inline library/maven]: Did you want to lose all these `ibmjava` and `alpine` tags?

```diff
-maven:3-ibmjava
-maven:3.6-alpine
-maven:3.6-ibmjava
-maven:3.6-ibmjava-alpine
-maven:3.6.0-alpine
-maven:3.6.0-ibmjava
-maven:3.6.0-ibmjava-alpine
-maven:alpine
-maven:ibmjava
-maven:ibmjava-alpine
```

Comment (tianon) [reply library/maven]: Friendly ping @carlossg -- was the removal of these tags intentional?  Should we still consider this PR for merging?

**Comment by carlossg** [comment]
Sorry, missed the comment, I have fixed the generation script and should have kept all the tags

**Comment by tianon** [comment]
<details>
<summary>Diff:</summary>

```diff
diff --git a/_bashbrew-arches b/_bashbrew-arches
index 07f6607..8f9ecd2 100644
--- a/_bashbrew-arches
+++ b/_bashbrew-arches
@@ -1,3 +1,7 @@
+maven:3-ibmjava-8 @ amd64
+maven:3-ibmjava-8 @ i386
+maven:3-ibmjava-8 @ ppc64le
+maven:3-ibmjava-8 @ s390x
 maven:3-jdk-7 @ amd64
 maven:3-jdk-7 @ arm32v5
 maven:3-jdk-7 @ arm32v7
@@ -12,6 +16,20 @@ maven:3-jdk-7-slim @ amd64
 maven:3-jdk-7-slim @ arm32v5
 maven:3-jdk-7-slim @ arm32v7
 maven:3-jdk-7-slim @ i386
+maven:3-jdk-8 @ amd64
+maven:3-jdk-8 @ arm32v5
+maven:3-jdk-8 @ arm32v7
+maven:3-jdk-8 @ arm64v8
+maven:3-jdk-8 @ i386
+maven:3-jdk-8 @ ppc64le
+maven:3-jdk-8 @ s390x
+maven:3-jdk-8-slim @ amd64
+maven:3-jdk-8-slim @ arm32v5
+maven:3-jdk-8-slim @ arm32v7
+maven:3-jdk-8-slim @ arm64v8
+maven:3-jdk-8-slim @ i386
+maven:3-jdk-8-slim @ ppc64le
+maven:3-jdk-8-slim @ s390x
 maven:3-jdk-10 @ amd64
 maven:3-jdk-10 @ arm32v5
 maven:3-jdk-10 @ arm32v7
@@ -26,20 +44,6 @@ maven:3-jdk-10-slim @ arm64v8
 maven:3-jdk-10-slim @ i386
 maven:3-jdk-10-slim @ ppc64le
 maven:3-jdk-10-slim @ s390x
-maven:3-jdk-11 @ amd64
-maven:3-jdk-11 @ arm32v5
-maven:3-jdk-11 @ arm32v7
-maven:3-jdk-11 @ arm64v8
-maven:3-jdk-11 @ i386
-maven:3-jdk-11 @ ppc64le
-maven:3-jdk-11 @ s390x
-maven:3-jdk-11-slim @ amd64
-maven:3-jdk-11-slim @ arm32v5
-maven:3-jdk-11-slim @ arm32v7
-maven:3-jdk-11-slim @ arm64v8
-maven:3-jdk-11-slim @ i386
-maven:3-jdk-11-slim @ ppc64le
-maven:3-jdk-11-slim @ s390x
 maven:3-jdk-12 @ amd64
 maven:3-jdk-12-alpine @ amd64
 maven:alpine @ amd64
@@ -48,10 +52,6 @@ maven:alpine @ arm64v8
 maven:alpine @ i386
 maven:alpine @ ppc64le
 maven:alpine @ s390x
-maven:ibmjava @ amd64
-maven:ibmjava @ i386
-maven:ibmjava @ ppc64le
-maven:ibmjava @ s390x
 maven:ibmjava-alpine @ amd64
 maven:latest @ amd64
 maven:latest @ arm32v5
diff --git a/_bashbrew-list b/_bashbrew-list
index 7dc10ce..5c17449 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -1,5 +1,4 @@
 maven:3
-maven:3-ibmjava
 maven:3-ibmjava-8
 maven:3-ibmjava-8-alpine
 maven:3-jdk-7
@@ -53,7 +52,6 @@ maven:3.6.0-jdk-12
 maven:3.6.0-jdk-12-alpine
 maven:3.6.0-slim
 maven:alpine
-maven:ibmjava
 maven:ibmjava-alpine
 maven:latest
 maven:slim
diff --git a/maven_ibmjava/Dockerfile b/maven_3-ibmjava-8/Dockerfile
similarity index 100%
rename from maven_ibmjava/Dockerfile
rename to maven_3-ibmjava-8/Dockerfile
diff --git a/maven_3-jdk-11-slim/mvn-entrypoint.sh b/maven_3-ibmjava-8/mvn-entrypoint.sh
similarity index 100%
rename from maven_3-jdk-11-slim/mvn-entrypoint.sh
rename to maven_3-ibmjava-8/mvn-entrypoint.sh
diff --git a/maven_3-jdk-11-slim/settings-docker.xml b/maven_3-ibmjava-8/settings-docker.xml
similarity index 100%
rename from maven_3-jdk-11-slim/settings-docker.xml
rename to maven_3-ibmjava-8/settings-docker.xml
diff --git a/maven_3-jdk-11-slim/Dockerfile b/maven_3-jdk-11-slim/Dockerfile
deleted file mode 100644
index ee03040..0000000
diff --git a/maven_3-jdk-11/Dockerfile b/maven_3-jdk-11/Dockerfile
deleted file mode 100644
index 5b8ade4..0000000
diff --git a/maven_slim/Dockerfile b/maven_3-jdk-8-slim/Dockerfile
similarity index 100%
copy from maven_slim/Dockerfile
copy to maven_3-jdk-8-slim/Dockerfile
diff --git a/maven_3-jdk-11/mvn-entrypoint.sh b/maven_3-jdk-8-slim/mvn-entrypoint.sh
similarity index 100%
rename from maven_3-jdk-11/mvn-entrypoint.sh
rename to maven_3-jdk-8-slim/mvn-entrypoint.sh
diff --git a/maven_3-jdk-11/settings-docker.xml b/maven_3-jdk-8-slim/settings-docker.xml
similarity index 100%
rename from maven_3-jdk-11/settings-docker.xml
rename to maven_3-jdk-8-slim/settings-docker.xml
diff --git a/maven_latest/Dockerfile b/maven_3-jdk-8/Dockerfile
similarity index 100%
copy from maven_latest/Dockerfile
copy to maven_3-jdk-8/Dockerfile
diff --git a/maven_ibmjava/mvn-entrypoint.sh b/maven_3-jdk-8/mvn-entrypoint.sh
similarity index 100%
rename from maven_ibmjava/mvn-entrypoint.sh
rename to maven_3-jdk-8/mvn-entrypoint.sh
diff --git a/maven_ibmjava/settings-docker.xml b/maven_3-jdk-8/settings-docker.xml
similarity index 100%
rename from maven_ibmjava/settings-docker.xml
rename to maven_3-jdk-8/settings-docker.xml
diff --git a/maven_latest/Dockerfile b/maven_latest/Dockerfile
index 32c2f3c..5b8ade4 100644
--- a/maven_latest/Dockerfile
+++ b/maven_latest/Dockerfile
@@ -1,10 +1,15 @@
-FROM openjdk:8-jdk
+FROM openjdk:11-jdk
 
 ARG MAVEN_VERSION=3.6.0
 ARG USER_HOME_DIR="/root"
 ARG SHA=fae9c12b570c3ba18116a4e26ea524b29f7279c17cbaadc3326ca72927368924d9131d11b9e851b8dc9162228b6fdea955446be41207a5cfc61283dd8a561d2f
 ARG BASE_URL=https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries
 
+# Maven fails with 'Can't read cryptographic policy directory: unlimited'
+# because it looks for $JAVA_HOME/conf/security/policy/unlimited but it is in
+# /etc/java-9-openjdk/security/policy/unlimited
+RUN ln -s /etc/java-11-openjdk /usr/lib/jvm/java-11-openjdk-$(dpkg --print-architecture)/conf
+
 RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
   && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
   && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha512sum -c - \
diff --git a/maven_slim/Dockerfile b/maven_slim/Dockerfile
index 6904378..ee03040 100644
--- a/maven_slim/Dockerfile
+++ b/maven_slim/Dockerfile
@@ -1,4 +1,4 @@
-FROM openjdk:8-jdk-slim
+FROM openjdk:11-jdk-slim
 
 ARG MAVEN_VERSION=3.6.0
 ARG USER_HOME_DIR="/root"
@@ -10,6 +10,11 @@ RUN apt-get update && \
       curl procps \
   && rm -rf /var/lib/apt/lists/*
 
+# Maven fails with 'Can't read cryptographic policy directory: unlimited'
+# because it looks for $JAVA_HOME/conf/security/policy/unlimited but it is in
+# /etc/java-9-openjdk/security/policy/unlimited
+RUN ln -s /etc/java-11-openjdk /usr/lib/jvm/java-11-openjdk-$(dpkg --print-architecture)/conf
+
 RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
   && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-${MAVEN_VERSION}-bin.tar.gz \
   && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha512sum -c - \
````

</details>

**Comment by carlossg** [comment]
@tianon so jdk-10 builds just need to be removed ?

**Comment by yosifkit** [comment]
Yeah, 9 and 10 were rapid releases and 11 is the new LTS. The wikipedia article on [java version history](https://en.wikipedia.org/wiki/Java_version_history) has the clearest "end of life" chart.

**Pull request merged**